core/src/main/java/io/substrait/relation/Set.java

```diff
@@ -42,7 +46,38 @@ public static Set.SetOp fromProto(SetRel.SetOp proto) {
 
   @Override
   protected Type.Struct deriveRecordType() {
-    return getInputs().get(0).getRecordType();
+    // The different inputs may have schemas that differ in nullability, but not in type.
+    // In that case we should return a schema that is nullable where any of the inputs is nullable.
```

Comment:
Looking at the docs for this (https://substrait.io/relations/logical_relations/#set-operation-types), the output nullability depends on which set operation is being performed.

Reply:
yep, I realized that as well but forgot to fix 😅 I'll try to tomorrow..